src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -1179,8 +1208,11 @@ private static void CollectErrors(List<ErrorDiagnostic> errors, ITypeReference r
             if (matches.Count == 1)
             {
                 // we have an exact match or a single ambiguous match
-                // return its type
-                return matches.Single().ReturnTypeBuilder(argumentSyntaxes);
+                var matchedOverload = matches.Single();
+                // record which function was used to get the type                
+                matchedFunctions.TryAdd(function, matchedOverload);
```

Comment:
I used here TryAdd - I was oddly getting duplicate inserts for this same symbol. Is this something that could happen? I thought one symbol can return only one type, and type assignment caching should prevent second call...

Reply:
I think some of the function symbols are created on a demand every time.

Reply:
ok, I need to dig a bit deeper then or perhaps change dictionary to use syntax as a key.

Reply:
@majastrz - I removed this part and exposed PlacementFlags through FunctionSymbol, similar to FunctionFlags